### PR TITLE
feat(client/server): bind shipping endpoint

### PR DESCRIPTION
**Pull request by DavidLiang01**
#

#### Shipping Delete Endpoint
- Allows for multiple deletions

#### Shipping Page
- Modal for creating a shipment
- Modified the change status button from toggle button to menu button
- Added row deletion/selection in table

#### Order Page
- When users place an order, it will also create a shipment

#### Sales Page
- When users place a sale, it will also create a shipment

## How to test this PR

- You need to test the PR locally because I modified the shipping endpoint 

## ClickUp reference

CU-q53c3d

## Notes for reviewers

- Click up [link](https://app.clickup.com/t/q53c3d)

## Checklist

- [x] I renamed the title of my pull request to follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [x] I have added tests to cover my changes (if applicable).

### CODE REVIEW

**Review by gordonpn — COMMENTED** [review]
Should making a sale create a shipping as well?

**Comment by DavidLiang01** [comment]
> Should making a sale create a shipping as well?

yes, I will add it

**Comment by Mackbellemore** [comment]
Task linked: [CU-q53c3d Connect Shipping endpoints with client side](https://app.clickup.com/t/q53c3d)

File: client/src/components/Shipping/ShippingTable.jsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
```
Comment (gordonpn) [inline client/src/components/Shipping/ShippingTable.jsx]: Can you use the `usePagination` hook?

File: client/src/components/Shipping/ShippingTable.jsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+import useShippingTable from 'hooks/useShippingTable.jsx';
+import { NoResultImage } from 'components/common/Image.jsx';
+import { formatDate } from 'utils/dateFunctions.js';
+import Loader from 'components/common/Loader.jsx';
+import Head from 'next/head';
+
+const ShippingHeader = () => (
+  <>
+    <Head>
+      <title>ERP - Orders</title>
```
Comment (gordonpn) [inline client/src/components/Shipping/ShippingTable.jsx]:
```suggestion
      <title>ERP - Shipping</title>
```

File: client/src/components/Shipping/ShippingTable.jsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  if (isSuccess && data.data.length === 0) {
+    return (
+      <>
+        <ShippingHeader />
+        <Heading size="xl" textAlign="center" mt={5}>
+          No shippings found.
+        </Heading>
+        <AddShipmentForm showButton={true} />
+
+        <NoResultImage />
```
Comment (gordonpn) [inline client/src/components/Shipping/ShippingTable.jsx]: I think we can remove this image for this page because the map takes a lot of space already.

**Review by sujvn — APPROVED** [review]
updated map is sick

**Review by gordonpn — APPROVED** [review]

**Pull request merged**